src/import-export/csv-imp/gnc-tokenizer.cpp

```diff
@@ -0,0 +1,101 @@
+#include "gnc-tokenizer.hpp"
+#include "gnc-csv-tokenizer.hpp"
+#include "gnc-dummy-tokenizer.hpp"
+#include "gnc-fw-tokenizer.hpp"
+
+#include <iostream>
+#include <fstream>      // fstream
+#include <vector>
+#include <string>
+#include <algorithm>    // copy
+#include <iterator>     // ostream_operator
+#include <memory>
+
+#include <boost/locale.hpp>
+
+extern "C" {
+#include <goffice/go-glib-extras.h>
+}
+
+std::unique_ptr<GncTokenizer> GncTokenizerFactory(GncImpFileFormat fmt)
+{
+    std::unique_ptr<GncTokenizer> tok(nullptr);
+    switch (fmt)
+    {
+    case GncImpFileFormat::CSV:
+        tok.reset(new GncCsvTokenizer());
+        break;
+    case GncImpFileFormat::FIXED_WIDTH:
+        tok.reset(new GncFwTokenizer());
+        break;
+    default:
+        tok.reset(new GncDummyTokenizer());
+        break;
+    }
+
+    return tok;
+}
+
+void
+GncTokenizer::load_file_generic(const std::string& path)
```

Comment:
Redid the overriding as you proposed and now it's working. I guess I missed a "virtual" declaration in GncTokenizer. Will be in the next push.